frontend/src/views/Lobby.tsx

```diff
@@ -215,7 +217,10 @@ function LobbyPage() {
     if (!socketConnected && currentRoomId && currentUser && currentUser.userId) {
       connectUserToRoom(currentRoomId, currentUser.userId);
     }
-  }, [socketConnected, connectUserToRoom, currentRoomId, currentUser]);
+    if (isActive) {
+      history.push('/game');
+    }
```

Comment:
nit: I would move this into its own useEffect block that depends only on `isActive`